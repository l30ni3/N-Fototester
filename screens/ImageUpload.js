import React, {useState, useRef, useEffect} from 'react';
import {SafeAreaView, StyleSheet, View, Image, Platform} from 'react-native';
import {
  Button,
  Divider,
  Icon,
  Layout,
  TopNavigation,
  TopNavigationAction,
  Spinner,
  Text,
} from '@ui-kitten/components';
import {launchCamera, launchImageLibrary} from 'react-native-image-picker';
import {RNCamera} from 'react-native-camera';
import Permissions from 'react-native-permissions';

const GLOBAL = require('./components/constants');

export const ImageUpload = ({navigation}) => {
  const [photo, setPhoto] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  const [itemId, setItemId] = useState(null);
  const [permission, setPermission] = useState('undetermined');

  const BackIcon = props => <Icon {...props} name="chevron-left-outline" />;
  const CameraIcon = props => <Icon {...props} name="camera" />;
  const UploadIcon = props => <Icon {...props} name="upload" />;
  const CloseIcon = props => <Icon {...props} name="close" />;
  const NextIcon = props => (
    <Icon {...props} name="arrow-ios-forward-outline" />
  );

  const navigateBack = () => {
    navigation.goBack();
  };

  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={navigateBack} />
  );

  useEffect(() => {
    Permissions.check('photo').then(response => {
      // Response is one of: 'authorized', 'denied', 'restricted', or 'undetermined'
      setPermission(response);
    });
  }, []);

  useEffect(() => {
    itemId
      ? navigation.navigate('Ergebnis', {
          itemId: itemId,
        })
      : null;
  }, [itemId]);

  const handleTakePhoto = () => {
    launchCamera({noData: true}, response => {
      if (response) {
        setPhoto(response);
      }
    });
  };

  const handleChoosePhoto = () => {
    const options = {
      mediaType: 'photo',
      includeBase64: true,
    };
    launchImageLibrary({options}, response => {
      if (response.didCancel) {
        console.log('User cancelled image picker');
        setPhoto(null);
      } else if (response.error) {
        console.log('ImagePicker Error: ', response.error);
        setPhoto(null);
      } else {
        setPhoto(response);
      }
    });
  };

  const handleDeletePhoto = () => {
    setPhoto(null);
  };

  const handleUploadPhoto = () => {
    setIsLoading(true);
    let localUri =
      Platform.OS === 'ios' ? photo.uri.replace('file://', '') : photo.uri;
    let filename = photo.fileName;
    let match = /\.(\w+)$/.exec(filename);
    let type = match ? `image/${match[1]}` : `image`;
    let formData = new FormData();
    formData.append('photo', {
      uri: localUri,
      name: filename,
      type: type,
    });
    fetch(`${GLOBAL.SERVER_URL}/api/upload`, {
      method: 'POST',
      body: formData,
      // headers: {
      //   Accept: 'application/json',
      //   'Content-Type': 'multipart/form-data',
      // },
    })
      .then(response => response.text())
      .then(response => {
        console.log('upload succes', response);
        setItemId(response);
        setIsLoading(false);
        setPhoto(null);
      })
      .catch(error => {
        console.log('upload error', error);
        setIsLoading(false);
      });
  };

  return (
    <SafeAreaView style={{flex: 1}}>
      <TopNavigation
        title="Neue Messung"
        alignment="center"
        accessoryLeft={BackAction}
      />
      <Divider />
      <Layout
        style={{
          flex: 1,
        }}>
        {!isLoading ? (
          <>
            {photo ? (
              <View style={styles.container}>
                <Image source={{uri: photo.uri}} style={styles.preview} />
              </View>
            ) : (
              <View style={styles.container}></View>
            )}
            <View
              style={{
                flexDirection: 'row',
                justifyContent: 'center',
              }}>
              {photo ? (
                <>
                  <Button
                    style={styles.button}
                    onPress={handleDeletePhoto}
                    accessoryRight={CloseIcon}>
                    Verwerfen
                  </Button>
                  <Button
                    style={styles.button}
                    onPress={handleUploadPhoto}
                    accessoryRight={NextIcon}>
                    Verwenden
                  </Button>
                </>
              ) : (
                <>
                  <Button
                    style={styles.button}
                    onPress={handleChoosePhoto}
                    accessoryRight={UploadIcon}>
                    Aus Gallerie
                  </Button>
                  <Button
                    style={styles.button}
                    onPress={handleTakePhoto}
                    accessoryRight={CameraIcon}>
                    Neues Foto
                  </Button>
                </>
              )}
            </View>
          </>
        ) : (
          <View style={styles.spinner}>
            <Spinner size="large" />
            <Text style={styles.spinnertext}>
              Die Bilddaten werden verarbeitet ...{' '}
            </Text>
          </View>
        )}
      </Layout>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: 'black',
  },
  preview: {
    flex: 1,
    justifyContent: 'center',
  },
  button: {
    margin: 20,
  },
  spinner: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  spinnertext: {
    color: 'white',
    margin: 20,
  },
});
