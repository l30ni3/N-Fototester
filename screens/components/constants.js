module.exports = {
  SERVER_URL: 'http://localhost:5000',
  CROP_DATA: ['Winterweizen', 'Sommerweizen', 'Mais'],
  VARIETY_DATA: ['Akteur', 'Alexander', 'Alfons', 'Alpha'],
  GROWTH_DATA: [
    'BBCH 20',
    'BBCH 21',
    'BBCH 22',
    'BBCH 23',
    'BBCH 24',
    'BBCH 25',
    'BBCH 26',
    'BBCH 27',
    'BBCH 28',
    'BBCH 29',
  ],
  SOWING_DATA: ['früh', 'normal', 'spät'],
  CAROUSEL_ITEMS: [
    {
      title: 'Item 1',
      text: 'Text 1',
    },
    {
      title: 'Item 2',
      text: 'Text 2',
    },
    {
      title: 'Item 3',
      text: 'Text 3',
    },
    {
      title: 'Item 4',
      text: 'Text 4',
    },
    {
      title: 'Item 5',
      text: 'Text 5',
    },
  ],
};
