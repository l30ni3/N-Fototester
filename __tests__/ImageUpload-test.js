/**
 * @format
 */

import 'react-native';
import React from 'react';
import {ImageUpload} from '../screens/ImageUpload';
import * as eva from '@eva-design/eva';
import {EvaIconsPack} from '@ui-kitten/eva-icons';
import {ApplicationProvider, IconRegistry} from '@ui-kitten/components';
import renderer from 'react-test-renderer';
jest.useFakeTimers();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({goBack: jest.fn()}),
  useRoute: () => ({
    params: {
      crop: 'CropValue',
      growth: 'GrowthValue',
      variant: 'VariantValue',
      replicate: 'ReplicateValue',
    },
  }),
}));
jest.mock('react-native-image-picker', () => ({
  launchImageLibrary: jest.fn(),
  launchCamera: jest.fn(),
}));
jest.mock('react-native-permissions', () => ({
  Permissions: jest.fn(),
}));

const mockedParams = {
  route: {
    params: {
      crop: 'CropValue',
      growth: 'GrowthValue',
      variant: 'VariantValue',
      replicate: 'ReplicateValue',
    },
  },
  navigation: '',
};

it('renders correctly', () => {
  const tree = renderer
    .create(
      <>
        <IconRegistry icons={EvaIconsPack} />
        <ApplicationProvider {...eva} theme={eva.light}>
          <ImageUpload {...mockedParams} />
        </ApplicationProvider>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
