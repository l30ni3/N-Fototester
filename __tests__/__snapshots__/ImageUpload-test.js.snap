// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "zIndex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#222B45"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "tintColor": "#222B45",
                    "width": 24,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4280429381,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="24"
                      matrix={
                        Array [
                          6.123233995736766e-17,
                          1,
                          -1,
                          6.123233995736766e-17,
                          24,
                          0,
                        ]
                      }
                      opacity={0}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M13.36 17a1 1 0 0 1-.72-.31l-3.86-4a1 1 0 0 1 0-1.4l4-4a1 1 0 1 1 1.42 1.42L10.9 12l3.18 3.3a1 1 0 0 1 0 1.41 1 1 0 0 1-.72.29z"
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          Neue Messung
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "zIndex": 1,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          undefined,
        ]
      }
    />
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3366FF",
              "borderColor": "#3366FF",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 20,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                },
              ]
            }
          >
            Aus Gallerie
          </Text>
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#FFFFFF"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 20,
                    "marginHorizontal": 10,
                    "tintColor": "#FFFFFF",
                    "width": 20,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4294967295,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="24"
                      matrix={
                        Array [
                          -1,
                          1.2246467991473532e-16,
                          -1.2246467991473532e-16,
                          -1,
                          24,
                          24,
                        ]
                      }
                      opacity={0}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="2"
                      matrix={
                        Array [
                          -1,
                          1.2246467991473532e-16,
                          -1.2246467991473532e-16,
                          -1,
                          24,
                          9.999999999999998,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      rx="1"
                      ry="1"
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="16"
                      x="4"
                      y="4"
                    />
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="2"
                      matrix={
                        Array [
                          6.123233995736766e-17,
                          1,
                          -1,
                          6.123233995736766e-17,
                          25,
                          -13,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      rx="1"
                      ry="1"
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="4"
                      x="17"
                      y="5"
                    />
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="2"
                      matrix={
                        Array [
                          6.123233995736766e-17,
                          1,
                          -1,
                          6.123233995736766e-17,
                          11,
                          1,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      rx="1"
                      ry="1"
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="4"
                      x="3"
                      y="5"
                    />
                    <RNSVGPath
                      d="M8 14a1 1 0 0 1-.8-.4 1 1 0 0 1 .2-1.4l4-3a1 1 0 0 1 1.18 0l4 2.82a1 1 0 0 1 .24 1.39 1 1 0 0 1-1.4.24L12 11.24 8.6 13.8a1 1 0 0 1-.6.2z"
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M12 21a1 1 0 0 1-1-1v-8a1 1 0 0 1 2 0v8a1 1 0 0 1-1 1z"
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3366FF",
              "borderColor": "#3366FF",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 20,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                },
              ]
            }
          >
            Neues Foto
          </Text>
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#FFFFFF"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 20,
                    "marginHorizontal": 10,
                    "tintColor": "#FFFFFF",
                    "width": 20,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4294967295,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="24"
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={0}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGCircle
                      cx="12"
                      cy="14"
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      r="1.5"
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M19 7h-3V5.5A2.5 2.5 0 0 0 13.5 3h-3A2.5 2.5 0 0 0 8 5.5V7H5a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3v-8a3 3 0 0 0-3-3zm-9-1.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5V7h-4zm2 12a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z"
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
